### PR TITLE
Ek add legalshield

### PR DESCRIPTION
**Pull request by ekump**

### CODE REVIEW

File: lib/passport-mock/factories/legalshield.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        is_precancel_status: false,
+        has_core_client_access: true
+      },
+    },
+    basic_profile: {
+      id: id,
+      login: (params.login || faker.internet.userName),
+      email: null,
+      first_name: null,
+      last_name: null,
```
Comment (ekump) [inline lib/passport-mock/factories/legalshield.js]: email, first_name, last_name come back from the API null in prod

File: lib/passport-mock/factories/legalshield.js [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+      is_member: ( typeof params.is_member === 'undefined' || params.is_member === null ? true : params.is_member ),
```
Comment (ekump) [inline lib/passport-mock/factories/legalshield.js]: `is_member` is probably going to default to true 99% of the time, so if it's not defined at all just default to true. Just doing `params.is_member || true` isn't going to work if `params.is_member` is specifically defined as false.

File: lib/passport-mock/factories/legalshield.js (hunk 1 above)
Comment (jonstorer) [inline lib/passport-mock/factories/legalshield.js]: aren't they empty strings?

File: lib/passport-mock/factories/legalshield.js [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+        },
+        sub_type: (params.sub_type || "Regular Family Plan"),
+        plan_display_name: "Family Plan",
+        is_individual_plan: false,
+        plan_code: (params.plan_code || utils.makeNumber(3)),
+        is_precancel_status: false,
+        has_core_client_access: true
+      },
+    },
+    basic_profile: {
```
Comment (jonstorer) [inline lib/passport-mock/factories/legalshield.js]: this is actually a different API endpoint and does not come back in passport at all

File: lib/passport-mock/factories/legalshield.js [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+  params.emergency_contact = params.emergency_contact || {};
+  params.access_control = params.access_control || {};
+
+  var id = (params.id || utils.makeNumber(10)) ;
+  var firstName = (params.firstName || faker.name.firstName());
+  var lastName = (params.lastName || faker.name.lastName());
+  var displayName = (params.displayName || firstName + ' ' + lastName);
+  var email = (params.email || utils.makeEmail(firstName, lastName));
+  var json = {
+    Member: {
```
Comment (jonstorer) [inline lib/passport-mock/factories/legalshield.js]: this key isn't a part of the API response, it's body is what's in the API response

File: lib/passport-mock/factories/legalshield.js (hunk 3 above)
Comment (ekump) [inline lib/passport-mock/factories/legalshield.js]: @jonstorer bleargh ok. I was just going off the token fetcher results.

File: lib/passport-mock/factories/legalshield.js [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+      first_name: firstName,
+      last_name: lastName,
+      membership_number: id,
+      day_time_phone_number: null,
+      home_phone_number: null,
+      cell_phone_number: (params.cellphone || faker.phone.phoneNumber()),
+      last_4_of_tax_id: (params.ssn || utils.makeNumber(4)),
+      date_of_birth: (params.dob || faker.date.past()),
+      fulfillment_partner: {
+        phone: (params.fulfillment_partner.phone || faker.phone.phoneNumber()),
```
Comment (jonstorer) [inline lib/passport-mock/factories/legalshield.js]: won't this blow up if the params doesn't have a `fulfillment_partner` property? you'd call a property on undefined.

File: lib/passport-mock/factories/legalshield.js [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+  params.emergency_contact = params.emergency_contact || {};
+  params.access_control = params.access_control || {};
+
+  var id = (params.id || utils.makeNumber(10)) ;
+  var firstName = (params.firstName || faker.name.firstName());
+  var lastName = (params.lastName || faker.name.lastName());
+  var displayName = (params.displayName || firstName + ' ' + lastName);
+  var email = (params.email || utils.makeEmail(firstName, lastName));
+  var json = {
+    id: id,
```
Comment (jonstorer) [inline lib/passport-mock/factories/legalshield.js]: this isn't part of the raw json response

File: lib/passport-mock/factories/legalshield.js [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+    account_type: "Member",
```
Comment (jonstorer) [inline lib/passport-mock/factories/legalshield.js]: this isn't a part of the raw json response

File: lib/passport-mock/factories/legalshield.js [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+    provider: "legalshield",
+    displayName: displayName,
+    name: {
+      givenName: firstName,
+      middleName: "",
+      familyName: lastName
+    },
+    emails: [
```
Comment (jonstorer) [inline lib/passport-mock/factories/legalshield.js]: ditto

**Comment by jonstorer** [comment]
lgtm, can you fix the whitespace so it's consistent

**Pull request closed without merging**